joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreator.scala

```diff
@@ -177,16 +177,49 @@ class AstCreator(
     callAst(callNode, args)
   }
 
-  private def astForGlobalStmt(stmt: PhpGlobalStmt): Ast = {
-    // This isn't an accurater representation of what `global` does, but with things like `global $$x` being possible,
-    // it's very difficult to figure out correct scopes for global variables.
-
-    val varsAsts = stmt.vars.map(astForExpr)
-    val code     = s"${PhpOperators.global} ${varsAsts.map(_.rootCodeOrEmpty).mkString(", ")}"
+  private def astForGlobalStmt(stmt: PhpGlobalStmt): List[Ast] = {
+    val surroundingIter    = scope.getSurroundingMethods.drop(1).iterator // drop first to ignore global method
+    val surroundingMethods = scope.getSurroundingMethods.dropRight(1)     // drop last to ignore innermost method
+
+    surroundingMethods.foreach { _ =>
+      val innerMethod = surroundingIter.next()
+      innerMethod.node match {
+        case inner: NewMethod =>
+          val methodRef = methodRefNode(stmt, s"${inner.fullName}(...)", inner.fullName, Defines.Any)
```

Comment:
- MethodRef needs to be in the CFG of the surrounding function at the position where the inner function is defined.
- Only one MethodRef shall be created. Your code would create multiple if there are multiple global statements. Add a test case for this.